### PR TITLE
Replace predictor, regressor, classifier

### LINKED ISSUE #352: Better terms for predictor, regressor, classifier throughout the course
This is probably too much jargon, also there is a source for confusion because regressors can also mean features used in a regression (similar thing for predictors) as mentioned in the scikit-learn glossary for example: https://scikit-learn.org/stable/glossary.html#term-predictor
> In statistics, "predictors" refers to features.

The terms we agreed on:
- predictor -> predictive model
- regressor -> regression model
- classifier -> classification model

### PR DESCRIPTION
**Pull request by ArturoAmorQ**
Fix #352 (Replace predictor, regressor, classifier for predictive model, regression model and classification model)

Fix wording related to replacement (for keeping readability)

Modify the glossary accordingly.

Fix couple of typos on the process.

### CODE REVIEW

File: jupyter-book/appendix/glossary.md [hunk 1]
```diff
@@ -46,22 +46,27 @@ is thus the black dotted line. This decision rule is used to
...
 
-### classifier
+### classification model
 
 A model used for [classification](#classification). These models handle
 [targets](#target-label-annotation) that contains discrete values such as
 `0`/`1` or `cat`/`dog`. For example in scikit-learn `LogisticRegression` or
-`HistGradientBoostingClassifier` are classifier
+`HistGradientBoostingClassifier` are classification model
 [classes](#target-label-annotation).
```
Comment (lesteve) [inline jupyter-book/appendix/glossary.md:59]: Hmmm actually this label is not correct here, can you fix it in a separate PR?

classes is meant here as an OOP (Object Oriented Programming) way not a classes=labels way which is what the link goes to. The fix is just to remove the link.

**Comment by lesteve** [comment]
I propose we merge https://github.com/INRIA/scikit-learn-mooc/pull/408 first. Feel free to review it by the way to make sure I haven't reverted some of your recent changes by mistake!

With a little bit of luck (🙃) you will get conflicts in this PR one #408 is merged and then you can get a crash course into merging conflicts in git (maybe you have learned this already, not sure ...)

File: jupyter-book/appendix/glossary.md (hunk 1 above)
Comment (lesteve) [inline jupyter-book/appendix/glossary.md:59]:
```suggestion
classes
```

Comment (lesteve) [reply jupyter-book/appendix/glossary.md:59]: I think if you accept this suggestion it should do what I want.

File: python_scripts/cross_validation_ex_01.py [hunk 2]
```diff
@@ -3,12 +3,12 @@
 #
 # The aim of this exercise is to make the following experiments:
 #
-# * train and test a support vector machine classifier through
+# * train and test a support vector machine classification model through
```
Comment (lesteve) [inline python_scripts/cross_validation_ex_01.py:6]: I feel like you can use just "model" in all this file.

support vector machine classification model sounds a bit too long

File: python_scripts/cross_validation_ex_03.py [hunk 3]
```diff
@@ -12,15 +12,15 @@
 data, target = load_iris(return_X_y=True, as_frame=True)
 
 # %% [markdown]
-# Create a decision tree classifier that we will use in the next experiments.
+# Create a decision tree classification model that we will use in the next experiments.
```
Comment (lesteve) [inline python_scripts/cross_validation_ex_03.py:15]:
```suggestion
# Create a `DecisionTreeClassifier` that we will use in the next experiments.
```

File: python_scripts/cross_validation_ex_03.py [hunk 4]
```diff
@@ -12,15 +12,15 @@
...
 
 # %%
 # Write your code here.
 
 # %% [markdown]
 # As a first experiment, use the utility
 # `sklearn.model_selection.train_test_split` to split the data into a train
-# and test set. Train the classifier using the train set and check the score
+# and test set. Train the classification model using the train set and check the score
```
Comment (lesteve) [inline python_scripts/cross_validation_ex_03.py:23]:
```suggestion
# and test set. Train the model using the train set and check the score
```

File: python_scripts/cross_validation_ex_04.py [hunk 5]
```diff
@@ -13,7 +13,7 @@
 
 # %% [markdown]
 # The first step is to create a model. Use a machine learning pipeline
-# composed of a scaler followed by a logistic regression classifier.
+# composed of a scaler followed by a logistic regression classification model.
```
Comment (lesteve) [inline python_scripts/cross_validation_ex_04.py:16]: Here too either the class name or just "logistic regression"

logistic regression classification model is super weird ...

File: python_scripts/cross_validation_grouping.py [hunk 6]
```diff
@@ -12,7 +12,7 @@
 
 # %% [markdown]
 # We will recreate the same model used in the previous exercise:
-# a logistic regression classifier with preprocessor to scale the data.
+# a logistic regression classification model with preprocessor to scale the data.
```
Comment (lesteve) [inline python_scripts/cross_validation_grouping.py:15]: Same as above "logistic regression classification model" is too weird

File: python_scripts/cross_validation_sol_01.py [hunk 7]
```diff
@@ -5,12 +5,12 @@
 #
 # The aim of this exercise is to make the following experiments:
 #
-# * train and test a support vector machine classifier through
+# * train and test a support vector machine classification model through
```
Comment (lesteve) [inline python_scripts/cross_validation_sol_01.py:8]: similar as before (I think?) support vector machine classification model sounds a bit weird

File: python_scripts/cross_validation_sol_01.py [hunk 8]
```diff
@@ -125,9 +125,9 @@
 
 # %% [markdown]
 # Looking at the curve, we can clearly identify the over-fitting regime of
-# the SVC classifier when `gamma > 1`.
+# the SVC classification model when `gamma > 1`.
```
Comment (lesteve) [inline python_scripts/cross_validation_sol_01.py:128]:
```suggestion
# the `SVC` when `gamma > 1`.
```

File: python_scripts/cross_validation_time.py [hunk 9]
```diff
@@ -54,7 +54,7 @@
     data, target, shuffle=True, random_state=0)
 
 # %% [markdown]
-# We will use a decision tree regressor that we expect to overfit and thus not
+# We will use a decision tree regression model that we expect to overfit and thus not
```
Comment (lesteve) [inline python_scripts/cross_validation_time.py:57]: comment as above (I think?)

File: python_scripts/ensemble_ex_03.py [hunk 10]
```diff
@@ -48,7 +48,7 @@
 # the model.
 
 # %% [markdown]
-# Repeat the experiment using a random forest instead of an AdaBoost regressor.
+# Repeat the experiment using a random forest instead of an AdaBoost regression model.
```
Comment (lesteve) [inline python_scripts/ensemble_ex_03.py:51]:
```suggestion
# Repeat the experiment using a random forest instead of an `AdaBoostRegressor`.
```

File: python_scripts/ensemble_ex_05.py [hunk 11]
```diff
@@ -15,7 +15,7 @@
 target *= 100  # rescale the target in k$
 
 # %% [markdown]
-# First, create a histogram gradient boosting regressor. You can set the
+# First, create a histogram gradient boosting regression model. You can set the
```
Comment (lesteve) [inline python_scripts/ensemble_ex_05.py:18]: Use the class name

**Review by lesteve — COMMENTED** [review]
OK I haven't gone through everything but I feel there are a few cases where this makes things a bit too long/convoluted, for example:
- logistic regression classification model
- decision tree classification model
- support vector machine classification model
- histogram gradient boosting classification model

Recommendations:
- use class name when appropriate decision tree classification model -> `DecisionTreeClassifier`
- use model (rather than regression model or classification model) when appropriate

**Comment by lesteve** [comment]
> OK I haven't gone through everything but I feel there are a few cases where this makes things a bit too long/convoluted, for example:

> -    logistic regression classification model
> -    decision tree classification model
> -    support vector machine classification model
> -    histogram gradient boosting classification model

> Recommendations:

> -    use class name when appropriate decision tree classification model -> DecisionTreeClassifier
> -    use model (rather than regression model or classification model) when appropriate

To be honest maybe wait for second opinion, this is a pain to update everything ...

**Comment by lesteve** [comment]
> To be honest maybe wait for second opinion, this is a pain to update everything ...

So the weak consensus from Tuesday meeting was that it was too much pain and micro decisions for not enough impact, so let's close this for now. We can always decide to revive it at one point in the future.
